test/libponyc/badpony.cc

```diff
@@ -1004,3 +1004,22 @@ TEST_F(BadPonyTest, TypeErrorDuringArrayLiteralInference)
     "    ExpectX[C].trigger([C; C].values())";
   TEST_ERRORS_1(src, "type argument is outside its constraint");
 }
+
+TEST_F(BadPonyTest,DisallowPointerAndMaybePointerInEmbeededType)
+{
+  // From issue #2596
+  const char* src =
+    "struct Ok"
```

Comment:
To make sure, this example compiles/errors correctly, just add \n at the end of each line of your example

Reply:
This is because the compiler tests use a reduced version of the `builtin` package in order to reduce testing time. You'll have to add the definition of `MaybePointer` to this string [here](https://github.com/ponylang/ponyc/blob/master/test/libponyc/util.cc#L23). Given what you're doing in the test, the following should be sufficient.

```pony
struct MaybePointer[A]
  new create(that: A) => compile_intrinsic
```